sklearn/feature_extraction/text.py

```diff
@@ -1274,17 +1274,23 @@ def inverse_transform(self, X):
         if sp.issparse(X):
             # We need CSR format for fast row manipulations.
             X = X.tocsr()
+
+            def get_non_zero(i):
+                return X[i, :].nonzero()[1]
         else:
             # We need to convert X to a matrix, so that the indexing
             # returns 2D objects
-            X = np.asmatrix(X)
+            X = np.asarray(X)
```

Comment:
Should we use `check_array` instead of `np.asarray`?

Reply:
The "downside" with `check_array` is that the only way to turn off the finite check is using a context manager:

https://github.com/scikit-learn/scikit-learn/blob/b94332434d0117e3d86407560a206d1c7bee1c81/sklearn/utils/validation.py#L83-L89

Looking through the code base, it looks like we are okay with this and call `check_array` on the input of `inverse_transform`. I updated this PR with using `check_array`.